.github/workflows/validate-cddl.yml

```diff
@@ -0,0 +1,39 @@
+name: Validate CDDL and Example
+
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - 'Documentation/corim_profile/*.cddl'
+      - 'Documentation/corim_profile/examples/*.diag'
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Install Ruby cddl and cbor-diag gems
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y ruby
+          gem install cddl
+          gem install cbor-diag
+
+      - name: Convert DIAG to CBOR
+        run: |
+          diag2cbor.rb Documentation/corim_profile/examples/ocp-safe-sfr-fw-example.diag > example.cbor
+
+      - name: Fetch latest upstream CoRIM CDDL
+        run: |
+          curl -L -o corim-base-08-upstream.cddl https://github.com/ietf-rats-wg/draft-ietf-rats-corim/releases/latest/download/corim-autogen.cddl
```

Comment:
Using `-08-` in the file name is bound to go out of sync as soon as the latest release is bumped.